synapse/appservice/scheduler.py

```diff
@@ -149,6 +150,15 @@ def enqueue_for_appservice(
             return
 
         if events:
+            # XXX: Special patch just for Gitter which we should remove after the import,
+            # https://github.com/matrix-org/synapse/pull/14729
+            #
+            # Ignore events that come from our own users. We probably already know about
+            # them and sent them ourself.
+            events = [
+                event for event in events if not self._hs.is_mine_id(event.sender)
+            ]
+
             self.queuer.queued_events.setdefault(appservice.id, []).extend(events)
```

Comment:
We should probably add another `if events` here? Otherwise we may be queuing an empty list which didn't seem possible before, and may cause unnecessary work somewhere down the line.

Reply:
Added